zebra-network/src/meta_addr/arbitrary.rs

```diff
@@ -17,9 +19,31 @@ impl MetaAddr {
             .boxed()
     }
 
-    pub fn alternate_node_strategy() -> BoxedStrategy<Self> {
+    /// Create a strategy that generates [`MetaAddr`]s in the
+    /// [`PeerAddrState::NeverAttemptedAlternate`] state.
+    pub fn alternate_strategy() -> BoxedStrategy<Self> {
+        (canonical_socket_addr(), any::<PeerServices>())
+            .prop_map(|(socket_addr, services)| MetaAddr::new_alternate(&socket_addr, &services))
+            .boxed()
+    }
+
+    /// Create a strategy that generates [`MetaAddr`]s which are ready for outbound connections.
+    ///
+    /// TODO: Generate all [`PeerAddrState`] variants, and give them ready fields.
+    ///       (After PR #2276 merges.)
+    pub fn ready_outbound_strategy() -> BoxedStrategy<Self> {
         canonical_socket_addr()
-            .prop_map(|address| MetaAddr::new_alternate(&address, &PeerServices::NODE_NETWORK))
+            .prop_filter_map("failed MetaAddr::is_valid_for_outbound", |address| {
+                // Alternate nodes use the current time, so they're always ready
+                //
+                // TODO: create a "Zebra supported services" constant
+                let meta_addr = MetaAddr::new_alternate(&address, &PeerServices::NODE_NETWORK);
+                if meta_addr.is_valid_for_outbound() {
```

Comment:
Create a specific strategy for valid and ready outbound peers.

This was the missing check in the previous code:
```
                if meta_addr.is_valid_for_outbound() {
```